extensions/notebook/src/dialog/managePackages/addNewPackageTab.ts

```diff
@@ -234,6 +227,66 @@ export class AddNewPackageTab {
 		});
 	}
 
+	private async fetchCondaPackage(packageName: string): Promise<PipPackageOverview> {
+		let condaExe = this.jupyterInstallation.getCondaExePath();
+		let cmd = `"${condaExe}" search --json ${packageName}`;
+		let packageResult: string;
+		try {
+			packageResult = await this.jupyterInstallation.executeBufferedCommand(cmd);
+		} catch (err) {
+			throw new Error(this.PackageNotFoundError);
+		}
+
+		if (packageResult) {
+			let packageJson = JSON.parse(packageResult);
+			if (packageJson) {
+				if (packageJson.error) {
+					throw new Error(packageJson.error);
+				}
+
+				let packages = packageJson[packageName];
+				if (Array.isArray(packages)) {
+					let allVersions = packages.filter(pkg => pkg && pkg.version).map(pkg => pkg.version);
+					let singletonVersions = new Set<string>(allVersions);
```

Comment:
i'm not following why creating the new Set is necessary. can you explain?

Reply:
There are a lot of duplicate versions in the conda search results, since it lists all the available packages from the various sources. The Set here just removes those duplicates.